File: notification/slack.py
import requests
from logger.log import *
from config.conf import *
def slack_notification(slack_message):
    try:
        if SLACK_NOTIFICATION:
            send_notification = requests.post(SLACK_WEBHOOK, json={"text": slack_message},
                                              headers={"Content-Type": "application/json"})
            slack_response = send_notification.text
            if slack_response == "ok":
                log_info("Slack message sent successfully")
            else:
                log_error("Error in sending slack message")
        else:
            return
    except Exception as error:
        log_error(f"Error in sending slack notification : {error}")

File: notification/slack.go
package notification

import (
	"bytes"
	"encoding/json"
	"io"
	"net/http"
	"os"

	"github.com/boringtools/git-alerts/logger"
)

func SlackNotification(content string) {

	jsonData := []byte("{'text': '" + content + "'}")
	json.Marshal(jsonData)

	slackHook := os.Getenv("SLACK_HOOK")
	request, error := http.NewRequest("POST", slackHook, bytes.NewReader(jsonData))
	request.Header.Set("Content-Type", "application/json; charset=UTF-8")

	if error != nil {
		logger.LogERR("Error in making HTTP request to slack")
	}

	client := &http.Client{}
	response, error := client.Do(request)
	if error != nil {
		panic(error)
	}

	defer response.Body.Close()

	body, _ := io.ReadAll(response.Body)
	sBody := string(body)

	if sBody == "ok" {
		logger.Log("Slack notification sent successfully")
	} else {
		logger.LogERR("Error in sending slack notification")
	}
}
